File: rating-service/src/main/java/com/example/rating_service/repo/UserRatingRepo.java
package com.example.rating_service.repo;

import com.example.rating_service.model.entity.UserRatingEntity;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserRatingRepo extends MongoRepository<UserRatingEntity, String> {
    Optional<UserRatingEntity> findByTargetUserId(UUID userId);
}

File: rating-service/src/main/kotlin/com/example/rating_service/repo/UserRatingRepo.kt
package com.example.rating_service.repo

import com.example.rating_service.entity.UserRatingEntity
import java.util.UUID
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRatingRepo : MongoRepository<UserRatingEntity, UUID> {
    fun findByUserId(userId: UUID): UserRatingEntity?
}